AppCenter/AppCenter/Internals/Channel/MSChannelUnitDefault.m

```diff
@@ -197,99 +197,98 @@ - (void)enqueueItem:(id<MSLog>)item flags:(MSFlags)flags {
-                   self.configuration.groupId, batchId, container.logs[i].sid,
-                   [(MSAbstractLog *)container.logs[i] serializeLogWithPrettyPrinting:YES]);
-      }
+  // Add to pending batches.
+  [self.pendingBatchIds addObject:container.batchId];
+  if (self.pendingBatchIds.count >= self.configuration.pendingBatchesLimit) {
+
+    // The maximum number of batches forwarded to the ingestion at the same time has been reached.
+    self.pendingBatchQueueFull = YES;
+  }
+
+  // Optimization. If the current log level is greater than
+  // MSLogLevelDebug, we can skip it.
+  if ([MSAppCenter logLevel] <= MSLogLevelDebug) {
+    NSUInteger count = [container.logs count];
+    for (NSUInteger i = 0; i < count; i++) {
+      MSLogDebug([MSAppCenter logTag], @"Sending %tu/%tu log, group Id: %@, batch Id: %@, session Id: %@, payload:\n%@", (i + 1), count,
+                 self.configuration.groupId, container.batchId, container.logs[i].sid,
+                 [(MSAbstractLog *)container.logs[i] serializeLogWithPrettyPrinting:YES]);
     }
+  }
 
-    // Notify delegates.
-    [self enumerateDelegatesForSelector:@selector(channel:willSendLog:)
-                              withBlock:^(id<MSChannelDelegate> delegate) {
-                                for (id<MSLog> aLog in logArray) {
-                                  [delegate channel:self willSendLog:aLog];
-                                }
-                              }];
+  // Notify delegates.
+  [self enumerateDelegatesForSelector:@selector(channel:willSendLog:)
+                            withBlock:^(id<MSChannelDelegate> delegate) {
+                              for (id<MSLog> aLog in container.logs) {
+                                [delegate channel:self willSendLog:aLog];
+                              }
+                            }];
 
-    // Forward logs to the ingestion.
-    [self.ingestion
-                sendAsync:container
-                authToken:tokenInfo.authToken
-        completionHandler:^(NSString *ingestionBatchId, NSHTTPURLResponse *response, __attribute__((unused)) NSData *data, NSError *error) {
-          dispatch_async(self.logsDispatchQueue, ^{
-            if ([self.pendingBatchIds containsObject:ingestionBatchId]) {
+  // Forward logs to the ingestion.
+  [self.ingestion sendAsync:container
+                  authToken:tokenArray[tokenIndex].authToken
+          completionHandler:^(NSString *ingestionBatchId, NSHTTPURLResponse *response, __unused NSData *data, NSError *error) {
+            dispatch_async(self.logsDispatchQueue, ^{
+              if (![self.pendingBatchIds containsObject:ingestionBatchId]) {
+                MSLogWarning([MSAppCenter logTag], @"Batch Id %@ not expected, ignore.", ingestionBatchId);
+                return;
+              }
+              BOOL succeeded = response.statusCode == MSHTTPCodesNo200OK;
 
               // Success.
```

Comment:
Delete this comment and `// Failure` below since the variable literally indicates the state now

Reply:
```suggestion
```